File: target_java_project/src/main/java/com/goodhelp/common/service/TimezoneHelper.java
package com.goodhelp.common.service;

import org.springframework.stereotype.Service;

import java.time.ZoneId;
import java.time.ZoneOffset;
import java.time.ZonedDateTime;

/**
 * Helper service for timezone operations.
 * Migrated from PHP: App\Common\Service\TimezoneHelper
 */
@Service
public class TimezoneHelper {
    
    public static final String DEFAULT_TIMEZONE = "Europe/Kiev";
    
    /**
     * Get a human-readable label for a timezone offset.
     * 
     * @param offsetSeconds Offset in seconds from UTC
     * @return Label like "UTC+2" or "UTC-5"
     */
    public String getLabelForOffset(int offsetSeconds) {
        return getLabelForOffset(offsetSeconds, null);
    }
    
    /**
     * Get a human-readable label for a timezone offset.
     * 
     * @param offsetSeconds Offset in seconds from UTC
     * @param timezoneName Optional timezone name to append
     * @return Label like "UTC+2 (Europe/Kiev)"
     */
    public String getLabelForOffset(int offsetSeconds, String timezoneName) {
        int hours = offsetSeconds / 3600;
        int minutes = Math.abs((offsetSeconds % 3600) / 60);
        
        StringBuilder label = new StringBuilder("UTC");
        if (hours >= 0) {
            label.append("+");
        }
        label.append(hours);
        
        if (minutes > 0) {
            label.append(":").append(String.format("%02d", minutes));
        }
        
        if (timezoneName != null && !timezoneName.isEmpty()) {
            label.append(" (").append(timezoneName).append(")");
        }
        
        return label.toString();
    }
    
    /**
     * Get the current offset for a timezone.
     * 
     * @param timezone Timezone identifier (e.g., "Europe/Kiev")
     * @return Offset in seconds
     */
    public int getOffsetForTimezone(String timezone) {
        ZoneId zoneId = ZoneId.of(timezone);
        ZoneOffset offset = ZonedDateTime.now(zoneId).getOffset();
        return offset.getTotalSeconds();
    }
    
    /**
     * Check if a timezone identifier is valid.
     * 
     * @param timezone Timezone identifier
     * @return true if valid
     */
    public boolean isValidTimezone(String timezone) {
        try {
            ZoneId.of(timezone);
            return true;
        } catch (Exception e) {
            return false;
        }
    }
}

File: original_php_project/src/Common/Service/TimezoneHelper.php
<?php

namespace App\Common\Service;

class TimezoneHelper
{
    public const DEFAULT_TIMEZONE  = 'Europe/Kiev';

    private \DateTimeZone $utcTimezone;

    public function __construct()
    {
        $this->utcTimezone = new \DateTimeZone('UTC');
    }

    public function toTzDateTime(string $datetime, \DateTimeZone $tz): \DateTime
    {
        $date = new \DateTime($datetime);
        $date->setTimezone($tz);

        return $date;
    }

    public function toTzDateTimeFromDatetime(\DateTime $datetime, \DateTimeZone $tz): \DateTime
    {
        $date = clone $datetime;
        $date->setTimezone($tz);

        return $date;
    }

    public function toUtcTzDateTime(string $datetime, \DateTimeZone $fromTz): \DateTime
    {
        $date = new \DateTime($datetime, $fromTz);
        $date->setTimezone($this->utcTimezone);

        return $date;
    }

    public function getLabelForOffset(int $offset, ?string $timezoneName = null): string
    {
        [$sign, $hours, $minutes] = $this->offsetToHoursAndMinutesAndSign($offset);
        $offsetLabel = "{$sign}{$hours}:{$minutes}";


        $label = "(UTC$offsetLabel)";
        if ($timezoneName) {
            $label .= " {$timezoneName}";
        }

        return $label;
    }

    private function offsetToHoursAndMinutesAndSign(int $offset): array
    {
        $minutes = $offset % 3600;
        $minutes = $minutes > 0 ? $minutes / 60 : '00';

        $hours = abs((int) ($offset / 3600));
        $hours = $hours < 10 ? "0{$hours}" : $hours;

        $sign = $offset >= 0 ? '+' : '-';

        return [$sign, $hours, $minutes];
    }
}